spec/controllers/foods_controller_spec.rb

```diff
@@ -1,159 +1,191 @@
-        post :create, {:food => valid_attributes}, valid_session
-        expect(assigns(:food)).to be_a(Food)
-        expect(assigns(:food)).to be_persisted
+    describe 'PATCH #update' do
+      it "should redirect to login" do
+        food = FactoryGirl.create(:food)
+        expect(post(:update, id: food.to_param, food: { name: "New Name" })).
+          to redirect_to(new_user_session_url)
       end
+    end
 
-      it "redirects to the created food" do
-        post :create, {:food => valid_attributes}, valid_session
-        expect(response).to redirect_to(Food.last)
+    describe "DELETE #destroy" do
+      it "should redirect to login" do
+        food = FactoryGirl.create(:food)
+        expect(delete(:destroy, id: food.to_param)).
+          to redirect_to(new_user_session_url)
       end
     end
+  end
 
-    context "with invalid params" do
-      it "assigns a newly created but unsaved food as @food" do
-        post :create, {:food => invalid_attributes}, valid_session
-        expect(assigns(:food)).to be_a_new(Food)
-      end
+  context "logged in" do
+    login_user
 
-      it "re-renders the 'new' template" do
-        post :create, {:food => invalid_attributes}, valid_session
-        expect(response).to render_template("new")
+    describe 'GET #index' do
+      it "assigns all users foods as @foods" do
+        user_food = FactoryGirl.create(:food, user: @current_user)
+        FactoryGirl.create(:food)
+        get :index
+        expect(assigns(:foods)).to contain_exactly(user_food)
       end
     end
-  end
 
-  describe "PUT #update" do
-    context "with valid params" do
-      let(:new_attributes) {
-        skip("Add a hash of attributes valid for your model")
-      }
+    describe 'GET #new' do
+      it "assigns a new food as @food" do
+        get :new
+        expect(assigns(:food)).to be_a_new(Food)
+      end
+    end
 
-      it "updates the requested food" do
-        food = Food.create! valid_attributes
-        put :update, {:id => food.to_param, :food => new_attributes}, valid_session
-        food.reload
-        skip("Add assertions for updated state")
+    describe 'GET #edit' do
+      it "raises an error if a non-user food is requested" do
+        food = FactoryGirl.create(:food)
+        expect { get(:edit, id: food.to_param) }.
+          to raise_error(ActiveRecord::RecordNotFound)
       end
 
       it "assigns the requested food as @food" do
-        food = Food.create! valid_attributes
-        put :update, {:id => food.to_param, :food => valid_attributes}, valid_session
+        food = FactoryGirl.create(:food, user: @current_user)
+        get :edit, id: food.to_param
         expect(assigns(:food)).to eq(food)
       end
+    end
 
-      it "redirects to the food" do
-        food = Food.create! valid_attributes
-        put :update, {:id => food.to_param, :food => valid_attributes}, valid_session
-        expect(response).to redirect_to(food)
+    describe 'POST #create' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.